File: Reverse Linked List.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def reverseList(self, head):
        curr = head
        prv = nex = None
        while curr:
            nex = curr.next
            curr.next = prv
            prv = curr
            curr = nex
        return prv
            

                
                
       
        

File: Reverse Linked List.cpp
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode *prv = nullptr, *next = nullptr, *curr = head;
        while(curr){
            next = curr->next;
            curr->next = prv;
            prv = curr;
            curr = next;
        }
        return prv;
   }
};